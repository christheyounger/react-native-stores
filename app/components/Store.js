import React from 'react';
import { FlatList, ActivityIndicator, Text, View  } from 'react-native';

export default class Store extends React.Component {
  render() {
    return (
      <View>
        <Text>- { this.props.name }</Text>
        <FlatList style={{paddingLeft: 10}}
          data={this.props.branches}
          renderItem={({item}) => <Store name={item.name} branches={item.branches} />}
          keyExtractor={({id}, index) => id.toString()}
        />
      </View>
    )
  }
}
