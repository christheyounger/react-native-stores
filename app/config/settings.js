export default settings = {
  apiUrl: 'http://localhost:8000/api/'
};
