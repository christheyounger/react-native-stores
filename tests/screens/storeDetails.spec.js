import 'isomorphic-fetch';
import React from 'react';
import { TextInput } from 'react-native';
import renderer from 'react-test-renderer';
import StoreDetails from '../../app/screens/storeDetails';
import ConfirmModal from '../../app/components/confirm';
import settings from '../../app/config/settings';
import Store from '../../app/components/Store';

jest.mock('ScrollView', () => require.requireMock('ScrollViewMock'))

describe('screen.store', function() {
    var navigation, wrapper;
    const store = {
        id: 1, 
        name: 'first store', 
        branches: [
            { id: 2, name: 'second store', branches: [] },
        ],
    };
    
    beforeEach(() => {
        jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({}));
        navigation = {
            getParam: () => 1,
            push: jest.fn()
        }
        wrapper = renderer.create(
            <StoreDetails
                getStore={jest.fn()}
                store={store}
                loading={false}
                navigation={navigation}
            />
        );
    });
    it('shows loader', () => {
        wrapper = renderer.create(
            <StoreDetails
                getStore={jest.fn()}
                store={store}
                loading
                navigation={navigation}
            />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
    test('renders correctly', function() {
        expect(wrapper.toJSON()).toMatchSnapshot();
        global.fetch.mockClear();
    });
    test('can delete a thing', function(done) {
        process.nextTick(() => {
            wrapper.root.findByType(ConfirmModal).props.handleClick();
            expect(global.fetch).toHaveBeenCalledWith(`${settings.apiUrl}stores/1`, {method: 'DELETE'});
            done();
        });
    });
    test('updates the name', function(done) {
        process.nextTick(() => {
            wrapper.root.findByType(TextInput).props.onSubmitEditing({nativeEvent: {text: 'steve'}});
            expect(global.fetch).toHaveBeenCalledWith(`${settings.apiUrl}stores/1`, {method: 'PUT', headers: {'Content-Type': 'application/json'}, body: '{"name":"steve"}'});
            done();
        });
    });
    test('links to children', function(done) {
        process.nextTick(() => {
            const item = store.branches[0];
            wrapper.root.findByType(Store).props.handleClick(item);
            expect(navigation.push).toHaveBeenCalledWith('StoreDetails', item);
            done();
        });
    });
});
