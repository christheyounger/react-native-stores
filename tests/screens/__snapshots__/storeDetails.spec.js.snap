// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen.store renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "fontSize": 24,
      }
    }
    underlineColorAndroid="transparent"
    value="first store"
  />
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            Delete
          </Text>
        </View>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={false}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "stretch",
              "flexDirection": "column",
              "marginTop": 22,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              Delete first store
            </Text>
            <Text
              style={
                Object {
                  "padding": 20,
                }
              }
            >
              Are you sure you wish to 
              Delete first store
              ?
            </Text>
            <View
              style={
                Object {
                  "alignItems": "stretch",
                  "flexDirection": "column",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    No, don't
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "red",
                        },
                      ]
                    }
                  >
                    Delete
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 21,
        }
      }
    >
      Branches
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "branches": Array [],
            "id": 2,
            "name": "second store",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              onPress={[Function]}
            >
              - 
              second store
            </Text>
            <View
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              style={
                Object {
                  "paddingLeft": 10,
                }
              }
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`screen.store shows loader 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
